File: lib/genetic_algorithm/selection_functions/SelectionFunction.py

class SelectionFunction():
    """
    Dummy selection function that always returns the first individual of the population
    """

    def select(self, population):
        """
        Selects an individual, in this case it always selects the first individual of the population
        :param population: <Population> The population of the genetic algorithm to select an individual from
        :return: <InputIndividual> The selected individual
        """
        return population.get_individuals()[0]

File: love_ga_wrapper/love_ga_wrapper/lib/genetic_algorithms/selection_functions/SelectionFunction.lua
local class = require("love_ga_wrapper.lib.classes.class")
local extend = require("love_ga_wrapper.lib.classes.extend")
--------------------------------------------------------------------------------------------------------

-- class: SelectionFunction
-- Dummy selection function that always returns the first individual of the population
local SelectionFunction = class(function(self)

end)

-- select: Population -> Individual
-- Selects an individual, in this case it always selects the first individual of the population
function SelectionFunction.select(self, population)
    local individuals = population:getIndividuals()
    return individuals[1]
end

return SelectionFunction
